snuba/state/cache/abstract.py

```diff
@@ -30,7 +32,7 @@ def set(self, key: str, value: TValue) -> None:
 
     @abstractmethod
     def get_readthrough(
-        self, key: str, function: Callable[[], TValue], timeout: int
+        self, key: str, function: Callable[[], TValue], timeout: int, timer: Timer,
```

Comment:
What about making this optional? 
The timer in the end is used to track metrics, so it could be up to the caller whether they are actually interested in tracking metrics since we package this class as generic.

Reply:
Did this in 2f3bd0b.

Reply:
@fpacifici Are you also suggesting making timer optional throughout the entire codebase or just here? It's required everywhere else.